Gems/AtomLyIntegration/CommonFeatures/Editor/Scripts/Materials/assign_material_to_all_entities.py

```diff
@@ -0,0 +1,63 @@
+# coding:utf-8
+#!/usr/bin/python
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+    Apply a hard-coded material to all meshes in the level
+"""
+
+import os
+
+import azlmbr.asset as asset
+import azlmbr.bus as bus
+import azlmbr.entity as entity
+import azlmbr.legacy.general as general
+import azlmbr.math as math
+import azlmbr.editor as editor
+import azlmbr.render as render
+
+###########################################################################
+# Main Code Block, runs this script as main (testing)
+# -------------------------------------------------------------------------
+if __name__ == '__main__':
+    """Run this file as main"""
+
+    
+    # Get the material we want to assign
+    material_asset_path = os.path.join("materials", "types", "debug", "debugvertexstreams.azmaterial")
+    material_asset_id = asset.AssetCatalogRequestBus(
+        bus.Broadcast, "GetAssetIdByPath", material_asset_path, math.Uuid(), False)
+    general.log(f"Assigning {material_asset_path} to all mesh and actor components in the level.")
+
+    type_ids_list = editor.EditorComponentAPIBus(bus.Broadcast, 'FindComponentTypeIdsByEntityType', ["Mesh", "Actor", "Material"], 0)
+    
+    mesh_component_type_id = type_ids_list[0]
+    actor_component_type_id = type_ids_list[1]
+    material_component_type_id = type_ids_list[2]
+    
+    # For each entity
+    search_filter = entity.SearchFilter()
+    all_entities = entity.SearchBus(bus.Broadcast, "SearchEntities", search_filter)
+    
+    for current_entity_id in all_entities:
```

Comment:
maybe log either each entity being updated or at least the total number of entities that were updated at the end

Reply:
I think the visual output speaks for itself, though that would be a good way to test this script using another hydra script as an automated test. I'll create a task for that.